substrate/cli/src/informant.rs

```diff
@@ -57,6 +60,14 @@ pub fn start<C>(service: &Service<C>, exit: ::exit_future::Exit, handle: TaskExe
 		} else {
 			warn!("Error getting best block information");
 		}
+
+		// get cpu usage and memory usage
+		sys.refresh_system();
+		let memory_usage = sys.get_used_memory() as f64 / sys.get_total_memory() as f64;
```

Comment:
paranoid mode on:
I wonder how robust this `sysinfo` crate is. Can it return `0` from `get_total_memory`? Or can `get_processor_list` return `[]`? It would be kinda pity if it panics on telemetry data acquistion.

Reply:
for the first question, even when `get_total_memory` returns 0, it's kinda OK, as it's been converted to a float, the final result would be NaN, it wouldn't panic though.

for the second question, I see your concern, but it seems the crate's document does just that. Hmm. I can add the length check before accessing the `[]` to prevent the possible panic here.